{{ cookiecutter.app_name }}/setup.py

```diff
@@ -0,0 +1,87 @@
+"""
+Install {{ cookiecutter.app_name }} via setuptools
+"""
+import sys
+
+from setuptools import setup, find_packages
+from setuptools.command.test import test as testcommand
+
+with open('test_requirements.txt') as test_reqs:
+    tests_require = test_reqs.readlines()  # pylint: disable=invalid-name
+
+
+class PyTest(testcommand):
+    """PyTest class to enable running `python setup.py test`"""
+    user_options = testcommand.user_options[:]
+    user_options += [
+        ('coverage', 'C', 'Produce a coverage report for {{ cookiecutter.app_name }}'),
+        ('pep8', 'P', 'Produce a pep8 report for {{ cookiecutter.app_name }}'),
+        ('pylint', 'l', 'Produce a pylint report for {{ cookiecutter.app_name }}'),
+    ]
+    coverage = None
+    pep8 = None
+    lint = None
+    test_suite = False
+    test_args = []
+
+    def initialize_options(self):
+        testcommand.initialize_options(self)
+
+    def finalize_options(self):
+        testcommand.finalize_options(self)
+        self.test_suite = True
+        self.test_args = []
+        if self.coverage:
+            self.test_args.append('--cov')
+            self.test_args.append('{{ cookiecutter.app_name }}')
+        if self.pep8:
+            self.test_args.append('--pep8')
+        if self.lint:
+            self.test_args.append('--lint')
+
+    def run_tests(self):
+        # import here, cause outside the eggs aren't loaded
+        import pytest
+        # Needed in order for pytest_cache to load properly
+        # Alternate fix: import pytest_cache and pass to pytest.main
+        import _pytest.config
+
+        plugin_manager = _pytest.config.get_plugin_manager()
+        plugin_manager.consider_setuptools_entrypoints()
+        errno = pytest.main(self.test_args)
+        sys.exit(errno)
+
+
+README = open('README.rst').read()
+
+setup(
+    name='{{ cookiecutter.app_name }}',
+    version='0.0.0',
+    license='AGPLv3',
+    author='MIT ODL Engineering',
+    author_email='odl-engineering@mit.edu',
+    url='http://github.com/mitodl/{{ cookiecutter.app_name }}',
+    description="{{ cookiecutter.description|replace("\"","\\\"") }}",
```

Comment:
I'm not familiar with this templating language. It _looks_ like jinja2, but I don't see jinja2 in the branch, nor in docker-compose. So I can't figure out how to play with this, but is there a way to mix single quotes and double-quotes to avoid all the backslash-escaping? The line `description="{{ cookiecutter.description|replace("\"","\\\"") }}",` is not valid Python, so I assume this is fixed by the template rendering prior to being executed and perhaps that templating doesn't allow for single-quotes.